util/no_os_circular_buffer.c

```diff
@@ -275,6 +278,10 @@ static int32_t no_os_cb_operation(struct no_os_circular_buffer *desc,
 		if (ret == -NO_OS_EOVERRUN)
 			sticky_overrun = true;
 
+		/* If no data is available return error */
+		if(!available_size)
+			return -1;
```

Comment:
let's try to be creative with these return codes (use matching ones from errno.h), they can be really helpful when debugging with printfs

Reply:
done